stl/inc/xstring

```diff
@@ -4738,16 +4723,15 @@ private:
         const size_type _Old_capacity = _Mypair._Myval2._Myres;
         const size_type _New_capacity = _Calculate_growth(_New_size);
         auto& _Al                     = _Getal();
-        const pointer _New_ptr        = _Al.allocate(_New_capacity + 1); // throws
+        const pointer _New_ptr        = _Allocate_from_al(_Al, _New_capacity); // throws
 
-        _Start_element_lifetimes(_Unfancy(_New_ptr), _New_capacity + 1);
         _Mypair._Myval2._Orphan_all();
         _ASAN_STRING_REMOVE(*this);
         _Mypair._Myval2._Mysize = _New_size;
         _Mypair._Myval2._Myres  = _New_capacity;
         _Fn(_Unfancy(_New_ptr), _New_size, _Args...);
-        if (_BUF_SIZE <= _Old_capacity) {
-            _Al.deallocate(_Mypair._Myval2._Bx._Ptr, _Old_capacity + 1);
+        if (_Old_capacity != _SMALL_STRING_CAPACITY) { // large string engaged

```

Comment:
Should we add internal checks in these cases(`_Old_capacity>=_SMALL_STRING_CAPACITY`, like in `_Large_string_engaged`)?